samza-core/src/main/scala/org/apache/samza/system/chooser/BootstrappingChooser.scala

```diff
@@ -196,12 +196,12 @@ class BootstrappingChooser(
           val systemStream = systemStreamPartition.getSystemStream
 
           updatedSystemStreams += systemStream -> (updatedSystemStreams.getOrElse(systemStream, 0) - 1)
-        }
 
-        // If the offset we just read is the same as the offset for the last
-        // message (newest) in this system stream partition, then we have read
-        // all messages, and can mark this SSP as bootstrapped.
-        checkOffset(systemStreamPartition, offset, OffsetType.NEWEST)
+          // If the offset we just read is the same as the offset for the last
```

Comment:
`NEWEST` offsetType denotes the offset at the tail-end of the system. 

If we receive a offset equal to NEWEST, then it will be sufficient to mark a bootstrap SSP as caught-up(the check doesn't have to be greater than or equal to). 

Instead of the following check in `checkOffset`: 
```java
    val comparatorResult = systemAdmin.offsetComparator(offset, offsetToCheck)
    if (comparatorResult != null && comparatorResult.intValue() >= 0) {
      ...
      ...
    }
```

I think it should be like this:
```java
    val comparatorResult = systemAdmin.offsetComparator(offset, offsetToCheck)
    if (comparatorResult != null && comparatorResult.intValue() == 0) {
      ...
      ...
    }
```

I think this should solve the problem reported in `SAMZA-1728`.

Do you see any issues in this?

Reply:
There could be systems where there could be two different topics: one for bootstrap and the other for change capture. When systemAdmin advertises it's newest offset, it would be that of change capture's latest offset, when the job started. There is this corner case: By the time we end up consuming all of bootstrap topic and we switch to change capture, the initially advertised offset could be out of retention. In that case, greater than would be required.

Reply:
I understand why we need this check. Makes sense.